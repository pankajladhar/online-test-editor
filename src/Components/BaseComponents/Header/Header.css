.Header {
  padding: 20px;
  color: white; }
  .Header header {
    display: flex; }
  .Header .Header__title {
    margin: 0;
    font-size: 25px;
    text-align: center;
    display: flex;
    justify-content: center;
    align-items: center;
    flex: 1; }
  .Header .Header__Logo {
    display: inline-block;
    width: 40px;
    height: 40px; }
