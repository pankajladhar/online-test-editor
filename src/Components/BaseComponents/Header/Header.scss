.Header {
  background-color: #222;
  padding: 20px;
  color: white;

  header {
    display: flex;
  }

  .Header__title{
    margin: 0;
    font-size: 25px;
    text-align: center;
    display: flex;
    justify-content: center;
    align-items: center;
    flex: 1;
  }

  .Header__Logo {
    display: inline-block;
    width: 40px;
    height: 40px;
  }
}