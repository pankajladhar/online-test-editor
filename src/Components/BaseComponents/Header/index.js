import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Navigation from './../Navigation';
import './Header.css'

class Header extends Component {
    render() {
        return (
            <div className="Header">
                <header className="container">
                    <div className="Header__Branding">
                        <Link to="/">
                            <img className="Header__Logo" src={this.props.logoUrl} alt={this.props.title} />
                            {this.props.title && <h1 className="Header__title">{this.props.title}</h1>}
                        </Link>
                    </div>
                    <Navigation />
                </header>
            </div>
        );
    }
}

Header.propTypes = {
    title: PropTypes.string.isRequired,
    logoUrl: PropTypes.string.isRequired
};

export default Header;