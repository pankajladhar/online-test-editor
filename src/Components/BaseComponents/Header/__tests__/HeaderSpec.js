import React from 'react';
import renderer from 'react-test-renderer';
import Header from './../index';

describe('Header', () => {
    let headerProps = {}
    beforeEach(() => {
        headerProps = {
            title : "some title",
            logoUrl : "some url"
        }
    });

    it('renders correctly', () => {
        let tree = renderer.create(
            <Header {...headerProps} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

});
	