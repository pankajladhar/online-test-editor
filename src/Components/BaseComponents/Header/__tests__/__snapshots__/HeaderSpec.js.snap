// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div
  className="Header"
>
  <header
    className="container"
  >
    <div
      className="Header__Branding"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt="some title"
          className="Header__Logo"
          src="some url"
        />
        <h1
          className="Header__title"
        >
          some title
        </h1>
      </a>
    </div>
    <div
      className="Navigation Navigation--Closed"
    >
      <nav>
        <ul>
          <li>
            <a
              href="/configure"
              onClick={[Function]}
            >
              Configure Quiz
            </a>
          </li>
          <li>
            <a
              href="/try"
              onClick={[Function]}
            >
              Try Existing 
            </a>
          </li>
        </ul>
      </nav>
      <button
        className="toggleButton"
        onClick={[Function]}
      >
        <span
          className="line"
        />
        <span
          className="line"
        />
        <span
          className="line"
        />
      </button>
    </div>
  </header>
</div>
`;
